### PR TITLE
Use Drupal 8 container via extension

### LINKED ISSUE #89: Update Component again and subclass / make use of Drupal 8's service container for this project :)
The service_container used here is now part of Drupal 8 as a component!

Now we need to re-integrate it with the Drupal 8 code.

### PR DESCRIPTION
**Pull request by fabianx-ai**
This fixes #89.

BC breaks:
#### Before:

``` php
$definition['services']['foo'] = array('alias' => 'bar');
$factory_service = array('factory_service' => 'x', 'factory_method' => 'get');
$factory_class = array('factory_class' => 'SomeClass', 'factory_method' => 'get');
```
#### After:

``` php
$definition['aliases']['foo'] = 'bar';
$factory_service = array('factory' => array('@x', 'get'));
$factory_service = array('factory' => 'SomeClass::get');
```

### CODE REVIEW

**Comment by drupol** [comment]
Ok, so, now at this point, what should we do ?

**Comment by fabianx-ai** [comment]
We need to fix the tests :)

File: lib/Drupal/Component/DependencyInjection/Container.php [hunk 1]
```diff
@@ -366,7 +366,7 @@ public function set($id, $service, $scope = ContainerInterface::SCOPE_CONTAINER)
    * {@inheritdoc}
    */
   public function has($id) {
-    return isset($this->services[$id]) || isset($this->serviceDefinitions[$id]);
+    return isset($this->aliases[$id]) || isset($this->services[$id]) || isset($this->serviceDefinitions[$id]) || array_key_exists($id, $this->services);
```
Comment (fabianx-ai) [inline lib/Drupal/Component/DependencyInjection/Container.php:369]: This needs to go upstream.

File: lib/Drupal/Component/DependencyInjection/PhpArrayContainer.php [hunk 2]
```diff
@@ -254,7 +254,7 @@ protected function resolveServicesAndParameters($arguments) {
       }
 
       // Resolve services.
-      if ($argument[0] === '@') {
+      if (isset($argument[0]) && $argument[0] === '@') {
```
Comment (fabianx-ai) [inline lib/Drupal/Component/DependencyInjection/PhpArrayContainer.php:257]: This needs to go upstream, too.

File: patches/php-array-dumper-fix-php-5.3.diff [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/Drupal/Component/DependencyInjection/Dumper/OptimizedPhpArrayDumper.php b/lib/Drupal/Component/DependencyInjection/Dumper/OptimizedPhpArrayDumper.php
+index 23290b6..e38719b 100644
+--- a/lib/Drupal/Component/DependencyInjection/Dumper/OptimizedPhpArrayDumper.php
++++ b/lib/Drupal/Component/DependencyInjection/Dumper/OptimizedPhpArrayDumper.php
+@@ -286,7 +286,7 @@ class OptimizedPhpArrayDumper extends Dumper {
+         $arguments = $this->dumpCollection($call[1]);
+       }
+ 
+-      $code[$key] = [$method, $arguments];
++      $code[$key] = array($method, $arguments);
```
Comment (fabianx-ai) [inline patches/php-array-dumper-fix-php-5.3.diff:10]: This cannot go upstream - unless with very muck luck.

File: tests/src/DependencyInjection/ContainerTest.php [hunk 4]
```diff
@@ -467,16 +438,18 @@ protected function getMockContainerDefinition() {
...
         (object) array(
-          'type' => 'service',
+          'type' => 'private_service',
           'value' => $private_service,
           'id' => 'private__' . $private_hash,
         ),
         '%some_config%'
       ),
     );
+    /*
```
Comment (fabianx-ai) [inline tests/src/DependencyInjection/ContainerTest.php]: commented out code found!

**Comment by fabianx-ai** [comment]
Waiting for tests to finish, then merge.

**Comment by fabianx-ai** [comment]
Waiting for scrutinizer

**Pull request merged**